.github/workflows/cd_frontend.yml

```diff
@@ -0,0 +1,153 @@
+name: 'CD - Frontend'
+
+on:
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches:
+      - master
+      - dev
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    name: Frontend - Deploy package
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      name: Check out repository
+
+    - run: |
+        # Current/target branches' names
+        branch=${GITHUB_REF#refs/heads/}
+        target_branch="${{ github.event.pull_request.base.ref }}"
+
+        # Extension Publishers
+        rc_publisher=retroextcd-rc
+        pr_publisher=retroextcd-prval
+
+        # Extension Prefix
+        # Remove all forward slashes to use for Extension Id
+        rc_ext_prefix=${branch//\//}
+        pr_branch=${{ github.event.pull_request.head.ref }}
+        if [[ "$pr_branch" ]]; then
+          pr_ext_prefix=${pr_branch//\//}
+        fi
+
+        # Test Organizations
+        rc_test_organization=retroextcd-rc
+        pr_test_organization=retroextcd-prval
+
+        if [[ "$branch" == master ]] || [[ "$branch" == dev ]]; then
+          echo "Setting config for Release Candidate (RC) environment"
+          echo "##[set-output name=is_rc;]true"
+          echo "##[set-output name=publisher;]$rc_publisher"
+          echo "##[set-output name=ext_prefix;]$rc_ext_prefix"
+          echo "##[set-output name=test_organization;]$rc_test_organization"
+        elif [[ "$target_branch" == master ]] || [[ "$target_branch" == dev ]]; then
+          echo "Setting config for PR validation"
+          echo "##[set-output name=is_pr;]true"
+          echo "##[set-output name=publisher;]$pr_publisher"
+          echo "##[set-output name=ext_prefix;]$pr_ext_prefix"
+          echo "##[set-output name=test_organization;]$pr_test_organization"
+        fi
+      name: Set up default variables
+      id: config
+
+    - if: steps.config.outputs.is_rc == 'true' || steps.config.outputs.is_pr == 'true'
+      run: |
+        npm install -g \
+          tfx-cli \
+          node-jq
+      name: Install tools used for deployment
+
+    - if: steps.config.outputs.is_rc == 'true' || steps.config.outputs.is_pr == 'true'
+      run: |
+        # Create environment.tsx from template
+        cp config/environment.tsx.template config/environment.tsx
```

Comment:
Note that this cp operation will be obsolete once Mohamed's PR goes in.

Reply:
Thanks, I'll pair with @mhassaninmsft to update CD workflow in his PR.